### PR TITLE
[content-service] Start snapshots from-other

### LINKED ISSUE #4894: Init command beeing run when opening snapshot
### Bug description

When sharing a snapshot the init command in the start task “prepare webserver” is beeing run when opening it. According to the doc I think this shouldn’t be? The init command in the start task “setup” isn’t run as expected.

.gitpod.yml:
```
image:
  file: .gitpod.Dockerfile

tasks:
  - name: setup
    init: bash ./gitpod/init.sh 
    command: cd ./environment/ && docker-compose up

    # to have its own terminal, npm is run from here, to guarantee its depencies are installed up to this moment, they're also installed from here
  - name: npm install/run
    openMode: split-right
    command: echo 'Waiting for webserver container to start...' &&
     until docker ps | grep -q "webserver"; do sleep 1; done &&
     docker exec -it $(docker ps | grep "webserver" | awk '{ print $1 }') 
     bash -c "cd /var/www/laravel && npm install && npm run watch"

  - name: prepare webserver
    openMode: split-right  
    init: echo 'Waiting for webserver container to start...' &&
     until docker ps | grep -q "webserver"; do sleep 1; done &&
     docker exec -it $(docker ps | grep "webserver" | awk '{ print $1 }') 
     bash -c "/var/www/laravel/scripts/init_gitpod.sh"
```

init_gitpod.sh
```

#!/bin/bash

cd /var/www/
rm -R html
ln -s /var/www/laravel/public /var/www/html

# beautyful chat
cd /var/www/laravel
composer install
git submodule sync --recursive
git submodule update --init --recursive

cd ./resources/vue-beautiful-chat
#cd /var/www/laravel/resources/vue-beautiful-chat
npm install

cd /var/www/laravel/
composer install

# give file access
chmod 777 -R storage/


php artisan migrate
php artisan cache:clear
php artisan queue:restart
php artisan db:seed

#copy gitpod debugger config
cp -a ./gitpod/docker-php-ext-xdebug.ini /usr/local/etc/php/conf.d/

exit
```

### Steps to reproduce

-

### Expected behavior

The init command is only run when creating the workspace as described in the doc.

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by csweichel**
to make the tasks behave like a workspace restart.

This change makes snapshots behave as [documented](https://www.gitpod.io/docs/config-start-tasks#start-a-snapshot).

fixes #4894 

## How to test
1. Open https://csweichel-init-command-beeing-run-4894.staging.gitpod-dev.com#snapshot/3c239bd3-d3ff-4399-9ffe-e419ac99858a
2. `cat /workspace/.gitpod/ready` should be `from-backup`

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4905#" title="Author self-approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by svenefftinge — APPROVED** [review]
works nicely :-)

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 87d67a272a869db8b6b53de0f2be88c33dd68fad</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4905#" title="Author self-approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4905#pullrequestreview-712456703" title="Approved">svenefftinge</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
